include/MySQL_Thread.h

```diff
@@ -0,0 +1,288 @@
+#ifndef __CLASS_MYSQL_THREAD_H
+#define __CLASS_MYSQL_THREAD_H
+#define ____CLASS_STANDARD_MYSQL_THREAD_H
+#include "proxysql.h"
+#include "cpp.h"
+
+
+#define MIN_POLL_LEN 8
+#define MIN_POLL_DELETE_RATIO  8
+
+
+#define ADMIN_HOSTGROUP	-2
+#define STATS_HOSTGROUP	-3
+
+
+static unsigned int near_pow_2 (unsigned int n) {
+  unsigned int i = 1;
+  while (i < n) i <<= 1;
+  return i ? i : n;
+}
+
+class ProxySQL_Poll {
+
+  private:
+  void shrink() {
+    unsigned int new_size=near_pow_2(len+1);
+    fds=(struct pollfd *)realloc(fds,new_size*sizeof(struct pollfd));
+    myds=(MySQL_Data_Stream **)realloc(myds,new_size*sizeof(MySQL_Data_Stream *));
+		last_recv=(unsigned long long *)realloc(last_recv,new_size*sizeof(unsigned long long));
+		last_sent=(unsigned long long *)realloc(last_sent,new_size*sizeof(unsigned long long));
+    size=new_size;
+  };
+  void expand(unsigned int more) {
+    if ( (len+more) > size ) {
+      unsigned int new_size=near_pow_2(len+more);
+      fds=(struct pollfd *)realloc(fds,new_size*sizeof(struct pollfd));
+      myds=(MySQL_Data_Stream **)realloc(myds,new_size*sizeof(MySQL_Data_Stream *));
+			last_recv=(unsigned long long *)realloc(last_recv,new_size*sizeof(unsigned long long));
+			last_sent=(unsigned long long *)realloc(last_sent,new_size*sizeof(unsigned long long));
+      size=new_size;
+    }
+  };
+
+  public:
+	int poll_timeout;
+	unsigned long loops;
+	StatCounters *loop_counters;
+  unsigned int len;
```

Comment:
Be careful with the indentation in here, it's inconsistent. In pull requests, there's usually a lot of trouble when you view the diff due to differences in whitespace like this. Most of the ProxySQL code seems to be indented with tabs, so I would just stick with that. When you commit code that is not indented correctly, other people's editors will fix it automatically, leading to code diffs that are not relevant.